Создал ветку sprint_7-solution-in-file-manager

ViktorSalk opened the pull request.

Viktor3642 committed:
1) Создал класс FileBackedTaskManager, позволяющий сохранять и загружать задачи, эпики и подзадачи в файл.
2) Создал enum с типами задач.
3) Создал ManagerSaveException пользовательское исключение для обработки ошибок сохранения:
4) Создал тесты по новой фиче: сохранение и загрузку пустого файла; сохранение нескольких задач;
загрузку нескольких задач.
5) Создал static void main(String[] args) в классе FileBackedTaskManager и реализовал тестовый сценарий.

afomkina commented on test/java/manager/FileBackedTaskManagerTest.java:
```diff
@@ -0,0 +1,71 @@
+
+class FileBackedTaskManagerTest {
+
+    private final File tempFile = new File("temp_tasks.csv");
```
Нужно использовать `File.createTempFile`

ViktorSalk replied:
Для тестирования методов, использовал File.createTempFile.

afomkina commented on .idea/misc.xml:
Сейчас в работе есть лишние файлы, которые генерируются средой разработки, чтобы от них избавиться в файл `.gitignore` нужно добавить следующие строки:

```
.idea
```

Затем нужно будет удалить лишние папки `.idea` [из репозитория](https://docs.github.com/ru/repositories/working-with-files/managing-files/deleting-files-in-a-repository)

ViktorSalk replied:
Добавил в .gitignore  .idea и удалил misc.xml из репозитория.

afomkina commented on src/main/java/manager/FileBackedTaskManager.java:
```diff
@@ -0,0 +1,200 @@
+        }
+    }
+
+    public static void main(String[] args) throws IOException {
```
По тз нужно реализовать следующий сценарий:

```
Заведите несколько разных задач, эпиков и подзадач.
Создайте новый FileBackedTaskManager-менеджер из этого же файла.
Проверьте, что все задачи, эпики, подзадачи, которые были в старом менеджере, есть в новом.
```

ViktorSalk replied:
Реализовал пользовательский сценарий по ТЗ: завел несколько разных задач, эпиков и подзадач, создал новый FileBackedTaskManager менеджер из этого же файла и проверил: совпадают ли задачи эпики и подзадачи в первом и новом менеджере.

Viktor3642 committed:
1) Для тестирования методов, которые работают с файлами, использовал функцию создания временных файлов File.createTempFile в FileBackedTaskManagerTest.
2) В классе FileBackedTaskManager в методе main реализовал пользовательский сценарий: завел несколько разных задач, эпиков и подзадач, создал новый FileBackedTaskManager менеджер из этого же файла и проверил: совпадают ли задачи эпики и подзадачи в новом и старом менеджере,
3) Добавил в .gitignore .idea чтобы избавится от лишних файлов генерирующихся средой разработки.

Viktor3642 committed:
1) Добавил разделитель для прохождения проверки стиля

ViktorSalk committed:
Delete .idea/misc.xml

Удаляю лишний файл генерирующийся средой разработки.
